File: algorithms/645. Set Mismatch/solution.cc
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        vector<int> counter(n + 1);
        int expect = n * (n + 1) / 2;
        int actual = accumulate(nums.begin(), nums.end(), 0);
        for ( int i : nums ) {
            if ( 1 == counter[i] )
                return {i, expect - actual + i};
            ++counter[i];
        }
    }
};

File: algorithms/645. Set Mismatch/solution.py
class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        # @ hy299792458
        s1, s2, n = sum(nums), sum(set(nums)), len(nums)
        return [s1 - s2, n * (n + 1) / 2 - s2]
